pkg/server/desthost_backend_manager.go

```diff
@@ -47,7 +47,7 @@ func (dibm *DestHostBackendManager) Backend(ctx context.Context) (Backend, error
 		bes, exist := dibm.backends[destHost]
 		if exist && len(bes) > 0 {
 			klog.V(5).InfoS("Get the backend through the DestHostBackendManager", "destHost", destHost)
-			return dibm.backends[destHost][0], nil
+			return dibm.backends[destHost][dibm.random.Intn(len(bes))], nil
```

Comment:
@jkh52 @relyt0925 
I think Tyler has right - in normal circumstances, it is possible that multiple agents are having the same agent identifier, e.g. `ipv4=172.30.2.31`.

Then the connections are in `dibm.backends["172.30.2.31"][]*backends{agent1, agent2, agent3}`.

But the destHost strategy gives only back the `[0]` element from the slice - so only agent1.

Thanks,
Adam

Reply:
@relyt0925 @andrewsykim @jkh52 
So I have checked, and yes, what @jkh52 has described is a valid point: if serverCount > 1, for short amount of time, it could happen that an already connected agent gets connected - but then disconnected. So it is a transient thing during initial setup. Of course it is more problematic when there is only one server, but the serverCount > 1.

So I think it is better to give back a random ID (see Tyler inital version), even if we know it could have some bad effect. In a well-configured setup, the issue cannot be a concern.

Or we could have another strategy.

Thanks,
Adam